src/model/relationship/RelationshipImpl.java

```diff
@@ -0,0 +1,190 @@
+package model.relationship;
+
+import model.GeneologyRules;
+import model.person.IPerson;
+
+import java.util.Random;
+
+public class RelationshipImpl implements IRelationship {
+  private double regard;
+  private double desire;
+  private IPerson p1;
+  private IPerson p2;
+  private int anniversaryYear;
+  private RelationshipType romanticType;
```

Comment:
Should this just be called type?

Reply:
The reason it's called `romanticType` is because it only ever really exists when it's a romantic type. The only times it's used are in the `progressRomanticType()` method, the `romanticToPlatonic()` method, and the `getType()` method, where the type is either `romanticType`, or generated from the regard if `romanticType` is null. It's kind of a crappy implementation, but it's wrapped up in the separation between platonic and romantic relationships, and I wanted to get something in before I changed my mind about that again. I'm definitely open to any and all suggestions you may have to do with that representation.

Reply:
Why can't the type change when the regard changes? If we want to avoid changing it to a platonic relationship on accident we could first check to see if it's romantic and then update it?

Reply:
You mean having just a `type` that gets set on creation, after the regard is generated, and updated in the `grow/lessenRegard()` methods, as well as the `progressRomanticType()` method? That makes sense.

Reply:
Yes that is what I meant.